silkworm/db/snapshots/bittorrent/web_seed_client.hpp

```diff
@@ -0,0 +1,82 @@
+/*
+   Copyright 2024 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <set>
+#include <string>
+#include <vector>
+
+#include <boost/asio/io_context.hpp>
+#include <boost/url/url.hpp>
+// Disable warnings raised during compilation of libtorrent
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wc++11-compat"
+#pragma GCC diagnostic ignored "-Wshadow"
+#pragma GCC diagnostic ignored "-Wsign-conversion"
+#include <libtorrent/torrent_info.hpp>
+#pragma GCC diagnostic pop
+
+#include <silkworm/infra/concurrency/task.hpp>
+
+#include "../config.hpp"
+#include "web_session.hpp"
+
+namespace silkworm::snapshots::bittorrent {
+
+using TorrentFileList = std::vector<std::string>;
+using TorrentsByProvider = std::map<boost::urls::url, TorrentFileList>;
+using TorrentInfo = lt::torrent_info;
+using TorrentInfoPtr = std::shared_ptr<TorrentInfo>;
+inline auto torrent_info_compare = [](const TorrentInfoPtr& lhs, const TorrentInfoPtr& rhs) {
+    return lhs->name() < rhs->name();
+};
+using TorrentInfoPtrList = std::set<TorrentInfoPtr, decltype(torrent_info_compare)>;
+
+class WebSeedClient {
+  public:
+    WebSeedClient(std::vector<std::string>&& url_seeds, const PreverifiedList& preverified);
+    WebSeedClient(std::unique_ptr<WebSession>&& web_session,
+                  std::vector<std::string>&& url_seeds,
+                  const PreverifiedList& preverified);
```

Comment:
```suggestion
    WebSeedClient(std::vector<std::string> url_seeds, const PreverifiedList& preverified);
    WebSeedClient(std::unique_ptr<WebSession> web_session,
                  std::vector<std::string> url_seeds,
                  const PreverifiedList& preverified);
```

I suggest to pass by value by default instead of move-only ref (unless we absolutely have to).
This is a convention I use in my code, and I'd like it to be a common codebase convention. I'll add it to the proposal doc for further discussion.